backstory/src/main/java/com/google/sps/images/VisionManager.java

```
@@ -0,0 +1,101 @@
+    this.imageAnnotatorClient = imageAnnotatorClient;
+  }
+
+  @Override
+  public List<AnnotatedImage> createAnnotatedImagesFromImagesAsByteArrays(
+      List<byte[]> imagesAsByteArrays) throws IOException {
+    List<AnnotatedImage> annotatedImages = new ArrayList<>();
+
+    for (byte[] rawImageData : imagesAsByteArrays) {
+      List<EntityAnnotation> labelAnnotations = detectLabelsFromImageBytes(rawImageData);
+      AnnotatedImage annotatedImage = new AnnotatedImage(rawImageData, labelAnnotations);
+      annotatedImages.add(annotatedImage);
+    }
+
+    return annotatedImages;
+  }
+
+  /**
+   * Creates and returns label annotations for the image represented in bytes, using the Vision API.
+   *
+   * @param bytes The raw image byte data for the image from which labels will be annotated.
+   * @return The list of label annotations related to the image, with each individual label being
+   *     represented as an EntityAnnotation object.
+   */
+  private List<EntityAnnotation> detectLabelsFromImageBytes(byte[] bytes) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+    List<EntityAnnotation> labels;
+
+    Image img = Image.newBuilder().setContent(ByteString.copyFrom(bytes)).build();
+    Feature feat = Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
+    AnnotateImageRequest request =
+        AnnotateImageRequest.newBuilder().addFeatures(feat).setImage(img).build();
+    requests.add(request);
+
+    // ImageAnnotatorClient will make the requests
+    BatchAnnotateImagesResponse response = imageAnnotatorClient.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    // For the MVP there is only one image in the batch response.
+    AnnotateImageResponse res = responses.get(0);
+
+    if (res.hasError()) {
+      System.out.format("Error: %s%n", res.getError().getMessage());
+      return new ArrayList<EntityAnnotation>();
```

Should this throw instead of swallowing the error? How will the user of this method know whether they should retry detection?

---

Updated.